### PR TITLE
NullPointerException on nodes call when there are deleted nodes in the result

### PR DESCRIPTION
**Pull request by szbaalint**
I've changed the lat and lon type from double to Double, so it won't throw an exception when there are no such fields from the result.

### CODE REVIEW

File: src/main/java/de/westnordost/osmapi/map/changes/MapDataChangesWriter.java [hunk 1]
```diff
@@ -149,8 +140,10 @@ private void writeElementAttributes(Element element) throws IOException
 	private void writeNodeContents(Node node) throws IOException
 	{
 		LatLon position = node.getPosition();
-		attribute("lat", position.getLatitude());
-		attribute("lon", position.getLongitude());
+		if (position != null) {
+			attribute("lat", position.getLatitude());
+			attribute("lon", position.getLongitude());
+		}
```
Comment (westnordost) [inline src/main/java/de/westnordost/osmapi/map/changes/MapDataChangesWriter.java]: The API will reply with an APIBadXMLError (bad request) error if one tries to create or update a node with no position.

So, as having no position at this point is an error anyway, a nullpointer is more descriptive here (and also fails earlier). Would you revert the changes you made for MapDataChangesWriter.java?

**Comment by westnordost** [comment]
Oi, nice find! And thank you for the pull request.

I would add a test case in MapDataParserTest that tests that parsing a deleted node throws no error and in MapDataChangesWriterTest that tests that writing a node without a position leads to an error. 

I wonder how deleted ways and relations look like, i.e. a way with no nodes and a relation with no members would throw nullpointer exceptions in the current implementation.

**Comment by szbaalint** [comment]
I've reverted the writer so it will throw NullPointerException when there is no position set. I've also added two tests for theses cases.

**Comment by szbaalint** [comment]
I've found an example for a deleted relation and it has no members at all, so it could also cause error:
http://api.openstreetmap.org/api/0.6/relations?relations=1
http://api.openstreetmap.org/api/0.6/ways?ways=101

**Comment by westnordost** [comment]
Very cool, thank you!

I will then also fix the no-members thing with ways and relations

**Pull request merged**